packages/block-library/src/image/edit.native.js

```diff
@@ -9,67 +9,99 @@ import RNReactNativeGutenbergBridge from 'react-native-gutenberg-bridge';
  */
 import { MediaPlaceholder, RichText, BlockControls } from '@wordpress/editor';
 import { Toolbar, ToolbarButton } from '@wordpress/components';
+import { Component } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import ImageSize from './image-size';
 
-export default function ImageEdit( props ) {
-	const { attributes, isSelected, setAttributes } = props;
-	const { url, caption } = attributes;
-
-	const onUploadPress = () => {
+class ImageEdit extends Component {
+	onUploadPress() {
 		// This method should present an image picker from
 		// the device.
 		//TODO: Implement upload image method.
-	};
+	}
 
-	const onMediaLibraryPress = () => {
+	onMediaLibraryPress() {
 		RNReactNativeGutenbergBridge.onMediaLibraryPress( ( mediaUrl ) => {
 			if ( mediaUrl ) {
-				setAttributes( { url: mediaUrl } );
+				this.props.setAttributes( { url: mediaUrl } );
 			}
 		} );
-	};
+	}
 
-	if ( ! url ) {
+	toolbarEditButton() {
```

Comment:
hi 👋 :) !  question here: what's the purpose (or advantage if any) of moving what before was a `const toolbarEditButton` into a method? It's my understanding that the method will produce a new object each time, as opposed to the `const` which cannot be re-bound (or fwiw, "re-assigned"). Maybe we needed to update something the object inside referenced?

Reply:
`toolbarEditButton` was `const` before because `ImageEdit` was a function. Now that `ImageEdit` was refactored as a class, we could keep `toolbarEditButton` as a const inside the `render` method or make it into a method as I did.

My reasoning was simply to have that code out of `render` and avoid having even more code inside it. But if there is a clear advantage of having it inside `render`, then we can always move it there. 👍

Reply:
Thanks for the explanation @etoledom ! That clarifies for me 👍